pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/DataSchemaSegmentPruner.java

```diff
@@ -15,111 +15,38 @@
- *
- *
+ * The <code>DataSchemaSegmentPruner</code> class prunes segment based on whether the all the querying columns exist in
+ * the segment schema.
  */
 public class DataSchemaSegmentPruner implements SegmentPruner {
 
-  private static final String COLUMN_KEY = "column";
-
   @Override
-  public boolean prune(IndexSegment segment, ServerQueryRequest queryRequest) {
-    BrokerRequest brokerRequest = queryRequest.getBrokerRequest();
-    Schema schema = segment.getSegmentMetadata().getSchema();
-
-    // Check filtering columns
-    if (brokerRequest.getFilterQuery() != null && !filterQueryMatchedSchema(schema, brokerRequest.getFilterQuery(),
-        brokerRequest.getFilterSubQueryMap())) {
-      return true;
-    }
-
-    // Aggregation queries
-    if (brokerRequest.isSetAggregationsInfo()) {
-      Set<String> columns = new HashSet<>();
-
-      // Add expressions in aggregation functions
-      for (AggregationInfo aggregationInfo : brokerRequest.getAggregationsInfo()) {
-        if (!aggregationInfo.getAggregationType().equalsIgnoreCase("count")) {
-          String expression = AggregationFunctionUtils.getColumn(aggregationInfo);
-          TransformExpressionTree.compileToExpressionTree(expression).getColumns(columns);
-        }
-      }
-
-      // Add expressions in group-by clause
-      if (brokerRequest.isSetGroupBy()) {
-        for (String expression : brokerRequest.getGroupBy().getExpressions()) {
-          TransformExpressionTree.compileToExpressionTree(expression).getColumns(columns);
-        }
-      }
-
-      // Check whether schema contains all columns
-      for (String column : columns) {
-        if (!schema.hasColumn(column)) {
-          return true;
-        }
-      }
-
-      return false;
-    }
+  public void init(@Nonnull Configuration config) {
+  }
 
-    // Check selection columns
-    if (brokerRequest.getSelections() != null) {
-      if (brokerRequest.getSelections().getSelectionColumns() != null) {
-        for (String columnName : brokerRequest.getSelections().getSelectionColumns()) {
-          if ((!columnName.equalsIgnoreCase("*")) && (!schema.hasColumn(columnName))) {
-            return true;
-          }
-        }
-      }
-      // Check columns to do sorting,
-      if (brokerRequest.getSelections().getSelectionSortSequence() != null) {
-        for (SelectionSort selectionOrder : brokerRequest.getSelections().getSelectionSortSequence()) {
-          if (!schema.hasColumn(selectionOrder.getColumn())) {
-            return true;
-          }
-        }
-      }
-      return false;
-    }
-    // unsupported query type.
-    return true;
+  @Override
+  public boolean prune(@Nonnull IndexSegment segment, @Nonnull ServerQueryRequest queryRequest) {
+    Set<String> ColumnsInSchema = segment.getSegmentMetadata().getSchema().getColumnNames();
+    return !(containsAll(ColumnsInSchema, queryRequest.getFilterColumns()) && containsAll(ColumnsInSchema,
```

Comment:
should we not just construct columnsInQuery and do one check against that?

Reply:
Done